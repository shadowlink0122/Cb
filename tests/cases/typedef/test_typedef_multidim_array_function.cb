// typedef配列の高度なテストケース
const int ROWS = 3;
const int COLS = 4;

typedef int[COLS] Row;

Row createRow(int start) {
    Row row;
    for (int i = 0; i < COLS; i++) {
        row[i] = start + i;
    }
    return row;
}

int main() {
    // 各行をテスト
    for (int r = 0; r < ROWS; r++) {
        Row row = createRow(r * 10);
        
        // 値の検証
        for (int c = 0; c < COLS; c++) {
            int expected = r * 10 + c;
            if (row[c] != expected) {
                println("ERROR: row[%d] should be %d, got %d", c, expected, row[c]);
                return 1;
            }
        }
        
        // 出力
        for (int c = 0; c < COLS; c++) {
            println("row[%d] = %d", c, row[c]);
        }
    }
    
    println("Typedef multidim array function test passed");
    return 0;
}
