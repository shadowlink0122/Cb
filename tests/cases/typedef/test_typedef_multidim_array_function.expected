row[0] = 0
row[1] = 1
row[2] = 2
row[3] = 3
row[0] = 10
row[1] = 11
row[2] = 12
row[3] = 13
row[0] = 20
row[1] = 21
row[2] = 22
row[3] = 23
Typedef multidim array function test passed
